src/transport/SessionManager.cpp

```diff
@@ -930,16 +890,17 @@ void SessionManager::SecureUnicastMessageDispatch(const PacketHeader & partialPa
     // Associate the secure session with the connection, if not done already.
     if (peerAddress.GetTransportType() == Transport::Type::kTcp)
     {
-        Transport::ActiveTCPConnectionState * sessionConn = secureSession->GetTCPConnection();
-        if (sessionConn == nullptr)
+        auto sessionConn = secureSession->GetTCPConnection();
+        if (sessionConn.IsNull())
         {
             secureSession->SetTCPConnection(ctxt->conn);
         }
         else
         {
             if (sessionConn != ctxt->conn)
             {
-                ChipLogError(Inet, "Data received over wrong connection %p. Dropping it!", ctxt->conn);
+                ChipLogError(Inet, "Unicast data received over %p for wrong connection %p. Dropping it!",
```

Comment:
See above.

Reply:
Same thing: the issue is that the new message is misleading in a way the old one was not.

Reply:
Fixed in follow-up